[CI] RolloverIT#testRolloverWithDateMath failed at midnight

The rollover test with the date pattern happened and failed at midnight.

```
1> [2018-02-19T11:59:59,017][INFO ][o.e.a.a.i.r.RolloverIT   ] [testRolloverWithDateMath]: before test
...
1> [2018-02-19T12:00:00,546][INFO ][o.e.a.a.i.r.RolloverIT   ] [RolloverIT#testRolloverWithDateMath]: cleaning up after test
```

The new index name was derived from the next day rather than today.

```
FAILURE 1.85s J2 | RolloverIT.testRolloverWithDateMath <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: "test-2018.02.19-000004"
   >      but: was "test-2018.02.20-000004"
   > 	at __randomizedtesting.SeedInfo.seed([DE29C74DF54CE353:5FEB5F63B28659FB]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.action.admin.indices.rollover.RolloverIT.testRolloverWithDateMath(RolloverIT.java:232)

```
CI: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=sles/695/console
Log: [testRolloverWithDateMath.txt](https://github.com/elastic/elasticsearch/files/1738671/testRolloverWithDateMath.txt)

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/rollover/RolloverIT.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/rollover/RolloverIT.java
@@ -274,6 +274,7 @@ public class RolloverIT extends ESIntegTestCase {
 
     public void testRolloverWithDateMath() {
         ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
+        assumeTrue("only works on the same day", now.plusMinutes(5).getDayOfYear() == now.getDayOfYear());
         String index = "test-" + DateFormatters.forPattern("YYYY.MM.dd").format(now) + "-1";
         String dateMathExp = "<test-{now/d}-1>";
         assertAcked(prepareCreate(dateMathExp).addAlias(new Alias("test_alias")).get());

